python/cudf/cudf/core/column/categorical.py

```diff
@@ -1526,6 +1527,28 @@ def _create_empty_categorical_column(
     )
 
 
+def _union_categoricals(
+    to_union: List[Union[cudf.Series, cudf.Index]],
+    sort_categories: bool = False,
+    ignore_order: bool = False,
+):
+
+    if ignore_order:
+        raise TypeError("ignore_order is not yet implemented")
+
+    result_col = CategoricalColumn._concat([obj._column for obj in to_union])
+    if sort_categories:
+        sorted_categories = (
+            cudf.Series(result_col.categories)
+            .sort_values(ascending=True, ignore_index=True)
+            ._column
+        )
+        result_col = result_col.cat().reorder_categories(
+            new_categories=sorted_categories
+        )
+    return cudf.Index(result_col)
```

Comment:
Can we maybe put this in `cudf.api.types` just to align it with where it will go in the future and add a `#TODO` to change it when `cudf.Categorical` lands?

Reply:
Done 👍